Sources/MongoKitten/CollectionHelpers/Collection+Update.swift

```diff
@@ -18,6 +18,18 @@ extension MongoCollection {
             )
         }.decodeReply(UpdateReply.self)._mongoHop(to: hoppedEventLoop)
     }
+
+    public func updateOne<E: Encodable>(
```

Comment:
I'd prefer naming these similarly to `insertEncoded` to allow for better discoverability instead. We also cannot call it the same as the other implementation. Since `Document` itself it `Encodable` this is almost guaranteed to cause issues for someone that inserts raw documents currently. Even if the current swift compilers know how to apply the right generics, history proves that Swift's complex generics system is prone to error.

Reply:
Agree!